### PR TITLE
Add hidden field in Tracks collection to signify which tracks to show on frontend

### LINKED ISSUE #13: Allow hidden tracks
`hidden` field on a track should mean it isn't exposed on the frontend. 
For ex, I want to hide this track 
Screenshot 2024-02-01 at 1 26 07 AM

### PR DESCRIPTION
**Pull request by daredevil0905**
Fixes #13 
## Summary
Added a field `hidden` in the `Track` interface which designates a track to be hidden on the front end if this field is set to `true`.

### CODE REVIEW

File: apps/backend/functions/src/routes/tracks.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ export const getTracks = onCall(async (request) => {
                 title: t.data().title,
                 image: t.data().image,
                 description: t.data().description,
-                problems: t.data().problems.map(x => x.id)
+                problems: t.data().problems.map(x => x.id),
```
Comment (hkirat) [inline apps/backend/functions/src/routes/tracks.ts:17]: These should be filtered from the backend itself. 
Also added a comment here - https://github.com/code100x/daily-code/issues/13#issuecomment-1921228116

File: packages/store/src/atoms/tracks.ts [hunk 2]
```diff
@@ -20,4 +20,5 @@ export interface Track {
     title: string;
     problems: string[];
     description: string;
+    hidden: boolean;
```
Comment (hkirat) [inline packages/store/src/atoms/tracks.ts:23]: Frontend should never know about this field

Comment (daredevil0905) [reply packages/store/src/atoms/tracks.ts]: Made the change

File: apps/backend/functions/src/routes/tracks.ts (hunk 1 above)
Comment (daredevil0905) [reply apps/backend/functions/src/routes/tracks.ts]: Made the change

**Pull request merged**